code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -491,38 +499,25 @@
 		UnregisterSignal(xeno, COMSIG_XENOMORPH_ATTACK_LIVING)
 	to_chat(xeno, span_xenonotice("You will now[xeno.vampirism ? "" : " no longer"] heal from attacking"))
 
-///called on regen, handles regen rate reduction
-/datum/action/xeno_action/vampirism/proc/on_regen(mob/living/carbon/xenomorph/dracula, list/heal_data)
-	SIGNAL_HANDLER
-	if(!leech_count)
-		return
-	//heals 10% extra per leeched
-	var/heal_mod = 1 + (leech_count/10)
-
-	heal_data[1] = (heal_data[1] * heal_mod)
-
 ///Adds the slashed mob to tracked damage mobs
-/datum/action/xeno_action/vampirism/proc/on_slash(datum/source, mob/living/target, damage, list/damage_mod, list/armor_mod)
+/datum/action/xeno_action/vampirism/proc/on_slash(datum/source, mob/living/target, damage)
 	SIGNAL_HANDLER
 	if(target.stat == DEAD)
 		return
 	if(!ishuman(target)) // no farming on animals/dead
 		return
-	if(last_leeched == target)
+	if(!COOLDOWN_CHECK(src, last_healed))
 		return
-	addtimer(CALLBACK(src, .proc/end_leech), VAMPIRISM_MOB_DURATION)
-	leech_count++
-	last_leeched = target
-	deltimer(clear_timer)
-	clear_timer = addtimer(CALLBACK(src, .proc/clear_leeched), VAMPIRISM_MOB_DURATION, TIMER_STOPPABLE)
+	var/mob/living/carbon/xenomorph/x = owner
+	x.adjustBruteLoss(-((x.xeno_caste.max_health - x.health) / 4))
+	x.adjustFireLoss(-((x.xeno_caste.max_health - x.health) / 4))
 	update_button_icon()
+	COOLDOWN_START(src, last_healed, heal_delay)
+	play_particles(x)
 
-///Called when the leech effect is supposed to end
-/datum/action/xeno_action/vampirism/proc/end_leech()
-	leech_count--
-	update_button_icon()
+/datum/action/xeno_action/vampirism/proc/play_particles(attacker)
+	particle_holder = new(attacker, /particles/xeno_slash/vampirism)
+	particle_holder.pixel_y = 18
+	particle_holder.pixel_x = 18
+	QDEL_IN(particle_holder, 2 SECONDS)
 
```

Comment:
this can bug out and due to tidi make it so the particles get deleted preemtively
also leaves a hanging ref so this will hard del